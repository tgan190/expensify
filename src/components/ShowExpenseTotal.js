import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expensesTotal';
import numeral from 'numeral';

export const ShowExpenseTotal = (props) => (
    <p>You are viewing {props.expenses.length} expenses totalling: <span>  {numeral(expensesTotal(props.expenses)/100.0).format('$0,0.00')}</span></p>
    );


const mapStateToProps = (state) => 
  ({
    expenses: selectExpenses(state.expenses, state.filters)
  });

export default connect(mapStateToProps)(ShowExpenseTotal);