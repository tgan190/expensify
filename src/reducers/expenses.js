// Expenses Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state,
        action.expense
      ];
    
    case 'REMOVE_EXPENSE':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_EXPENSE':
          
      // console.log('action',action);
      return state.map((expense) => {
        if (expense.id === action.id) {
           // console.log ('updating  redux...matched');
          return {
            ...expense,
            ...action.updates
          };
        } else {
            // console.log ('updating failed redux...no match');
          return expense;
        };
      });
    case 'SET_EXPENSES':
      return action.expenses;
    default:
      return state;
  }
};
