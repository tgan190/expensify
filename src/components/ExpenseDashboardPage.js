import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ShowExpenseTotal from './ShowExpenseTotal';

const ExpenseDashboardPage = () => (
  <div>
    <ShowExpenseTotal />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
