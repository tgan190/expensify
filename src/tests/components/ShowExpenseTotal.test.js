import React from 'react';
import { shallow } from 'enzyme';
import {ShowExpenseTotal} from '../../components/ShowExpenseTotal';
import expenses from '../fixtures/expenses';

test('should show count and total expenses for one expense', () => {
    const wrapper = shallow(<ShowExpenseTotal expenses={[expenses[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should show count and total expenses for 3 expenses', () => {
    const wrapper = shallow(<ShowExpenseTotal expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});