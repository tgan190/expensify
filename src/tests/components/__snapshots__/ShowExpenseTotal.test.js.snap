// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show count and total expenses for 3 expenses 1`] = `
<p>
  You are viewing 
  3
   expenses totalling: 
  <span>
      
    $1,141.95
  </span>
</p>
`;

exports[`should show count and total expenses for one expense 1`] = `
<p>
  You are viewing 
  1
   expenses totalling: 
  <span>
      
    $1.95
  </span>
</p>
`;
